atlas/readme.md

```diff
@@ -0,0 +1,325 @@
+
+### Provide forge with canonical representations of graph structures
+
+Two pieces to this:
+
+**Input.** In order to compose a schema, the compiler needs to reference other existing compiled graphs. This graph structure data needs to be represented in some way for transmission and caching.
+
+**Output.** The compiler needs to emit compiled graphs in some form.
+
+In other words, we an interchange format for graph structure data. We're being vague as to how right now—it could be a `.gql` file with directive, it could be some other format. Regardless, the atlas is responsible for defining what this is.
+
+### Provide flexible mechanisms to externally tag graph structures
+
+That is, we will want to mark pieces of the graph in arbitrary ways. For example, an organization may want to mark certain fields as containing personally identifying information. Naturally, you can do this with a custom directive on the field:
+
+```graphql
+type User {
+  address: String @userdata(privacy: PII)
+}
+```
+
+But constellation also allows us to do this in another schema entirely:
+
+```graphql
+extend type users.User {
+  users.address: String @userdata(privacy: PII)
+}
+```
+
+Here we're using hypothetical starql syntax, but it should be possible for such post-schema directives to be applied by other tooling as well.
+
+## Structure
+
+### Graphs are made of Points
+
+We use the term *point* to refer to all the stuff in a graph. These are all points:
+- types
+- inputs
+- interfaces
+- fields
+- unions
+- enums
+- scalars
+- directives
+- schemas
+
+### `atlas:` urls uniquely address points
+
+Technically atlas URLs are URIs or maybe URNs but [WHATWG basically thinks the distinction is unnecessary and confusing](https://url.spec.whatwg.org/#goals) so we'll just follow their lead and call them URLs.
+
+They take this form:
+
+```
+atlas:specifier:path?layer
+```
+
+Every part except the initial `atlas:` is optional.
+
+Some examples:
+
+```
+# these are all in the main atlas
+
+## a schema
+atlas::spotify
+
+## a point within the schema
+atlas::spotify.User
+
+## a field
+### note: still thinking about the syntax here
+atlas::spotify.id/spotify.User=spotify.ID
+
+## a scalar
+atlas::genius.ID
+
+## an extension field
+atlas::lyrical.geniusId/spotify.Song=genius.ID
+
+# these are in different atlases
+## identified by aliases
+atlas:corp-internal:svc.secrets/User=String
+
+## identified by atlas id
+atlas:[3deff29adabade98f3234acb9320]:svc.secrets/User=String
+```
+
+#### Atlas URLs start off by specifying the atlas
+
+Atlases can be specified either by *id* or *alias*. IDs are specified in `[brackets]`, aliases are bare words:
+
+Specified by id:
```

Comment:
What is the use case for specifying by id? That isn't something we would expect people to author?